### PR TITLE
It's easier to beg for forgiveness than to ask for permission

### PR DESCRIPTION
**Pull request by akaIDIOT**
Fixes race conditions and other confusing bugs by just trying rather than checking.

### CODE REVIEW

File: confidence/io.py [hunk 1]
```diff
@@ -251,14 +251,19 @@ def loadf(*fnames: typing.Union[str, PathLike],
...
             with open(fname, 'r') as fp:
                 LOG.info(f'reading configuration from file {fname}')
                 # default to empty dict, yaml.safe_load will return None for an empty document
                 return yaml.safe_load(fp.read()) or {}
-        else:
-            return default
+        except IOError:
+            # file does not exist or inaccessible
+            if default is NoDefault:
+                # no explicit default provided, continue original error
```
Comment (bartbroere) [inline confidence/io.py:262]: Nothing wrong with the original error :ok_hand:

**Review by bartbroere — APPROVED** [review]

File: confidence/io.py (hunk 1 above)
Comment (akaIDIOT) [reply confidence/io.py:262]: It is just so original, you know.

**Comment by akaIDIOT** [comment]
Still not too happy with the error simulator in tests, gotta find something better.

**Comment by akaIDIOT** [comment]
- [ ] Requires entry in changelog too

**Pull request merged**